gpiozero/exc.py

```diff
@@ -166,3 +166,5 @@ class PinFactoryFallback(PinWarning):
 class PinNonPhysical(PinWarning):
     "Warning raised when a non-physical pin is specified in a constructor"
 
+class ThresholdOutOfRange(GPIOZeroWarning):
```

Comment:
Hmmm, interesting. Should the `ValueError`s that you're raising elsewhere in this PR also be subclasses of `GPIOZeroError`? ping @waveform80

Reply:
Probably - yes. I have a feeling there are more non-gpiozero exceptions lying around, which we should clean up.